File: services/networkx_service/src/process/pubsub/selector_algorithm_publisher.py
import os
import json
from .publisher import Publisher
from network.envelope import Envelope


class SelectorAlgorithmPublisher(Publisher):
    def __init__(self, file_id, algorithm_name):
        super().__init__(os.environ["SELECTOR_ALGORITHMS_TOPIC"], '', 'fanout')
        self.file_id = file_id
        self.algorithm_name = algorithm_name

    def send(self, result):
        envelope = Envelope.send_selector_algorithm_result(self.file_id, self.algorithm_name, result)
        super().send(json.dumps(envelope), 0)

File: services/jgrapht_service/src/process/src/main/java/process/pubsub/SelectorAlgorithmPublisher.java
package process.pubsub;

import process.network.Envelope;

public class SelectorAlgorithmPublisher extends Publisher {
    private final String fileId;
    private final String algorithmName;

    public SelectorAlgorithmPublisher(String fileId, String algorithmName) {
        super(System.getenv("SELECTOR_ALGORITHMS_TOPIC"), "", "fanout");
        this.fileId = fileId;
        this.algorithmName = algorithmName;
    }

    public void send(String result) {
        String envelope = Envelope.sendSelectorAlgorithmResult(fileId, algorithmName, result);
        super.send(envelope);
    }
}
